### PR TITLE
increase memory of anndata_manipulate

### PR DESCRIPTION
**Pull request by pavanvidem**

### CODE REVIEW

**Comment by sanjaysrikakulam** [comment]
@pavanvidem Are you sure about this? Do we know how large the `input_size` can be?

**Comment by pavanvidem** [comment]
Currently, the tools is failing to concatenate 2 anndata objects of 2GB each.

**Comment by pavanvidem** [comment]
Is the default memory ~3.8G?

**Comment by sanjaysrikakulam** [comment]
The default from TPV shared DB is `cores: 4`, and mem will be lazily evaluated from our TPV config to `mem: cores * 3.8`, so basically, we are assigning `~15 GB` of memory.

Now, by overwriting this, we might not cover all possible scenarios. We must cover a minimum and maximum and your case of `input_size * 4`.

**Comment by pavanvidem** [comment]
Then `input_size * 4` is only a bit larger than default. I will write a case for larger files only.

**Comment by sanjaysrikakulam** [comment]
Something like this might work `min(max(input_size*4, 3.8*cores), 128)`

**Comment by pavanvidem** [comment]
all those tools need only 1 core

File: files/galaxy/tpv/tools.yml [hunk 1]
```diff
@@ -1374,3 +1374,28 @@ tools:
...
+  toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/.*:
+    cores: 1
+    mem: 4
+    rules:
+      - id: large_anndata_input
+        if: input_size > 2
+        cores: 1
+        mem: 20
+
+  toolshed.g2.bx.psu.edu/repos/iuc/anndata_import/anndata_manipulate/.*:
```
Comment (sanjaysrikakulam) [inline files/galaxy/tpv/tools.yml:1387]:
```suggestion
  toolshed.g2.bx.psu.edu/repos/iuc/anndata_import/anndata_import/.*
```

File: files/galaxy/tpv/tools.yml [hunk 2]
```diff
@@ -1374,3 +1374,28 @@ tools:
...
+    cores: 1
+    mem: 4
+
+  toolshed.g2.bx.psu.edu/repos/iuc/anndata_export/anndata_manipulate/.*:
```
Comment (sanjaysrikakulam) [inline files/galaxy/tpv/tools.yml:1391]:
```suggestion
  toolshed.g2.bx.psu.edu/repos/iuc/anndata_export/anndata_export/.*:
```

File: files/galaxy/tpv/tools.yml [hunk 3]
```diff
@@ -1374,3 +1374,28 @@ tools:
...
+    cores: 1
+    mem: 4
+
+  toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_manipulate/.*:
```
Comment (sanjaysrikakulam) [inline files/galaxy/tpv/tools.yml:1395]:
```suggestion
  toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/.*:
```

File: files/galaxy/tpv/tools.yml [hunk 4]
```diff
@@ -1374,3 +1374,28 @@ tools:
...
+    cores: 1
+    mem: 4
+
+  toolshed.g2.bx.psu.edu/repos/iuc/modify_loom/anndata_manipulate/.*:
```
Comment (sanjaysrikakulam) [inline files/galaxy/tpv/tools.yml:1399]:
```suggestion
  toolshed.g2.bx.psu.edu/repos/iuc/modify_loom/modify_loom/.*:
```

**Review by sanjaysrikakulam — CHANGES_REQUESTED** [review]

**Review by sanjaysrikakulam — APPROVED** [review]

**Pull request merged**